shoop/core/cache.py

```diff
@@ -0,0 +1,161 @@
+# -*- coding: utf-8 -*-
+"""
+Utilities for caching, with versioned caching and automatic timeout determination.
+
+Versioning works by way of "namespaces". Namespaces are the first colon-separated part
+of cache keys.
+
+For instance, the cache keys `price:10`, `price:20`, and `price` all belong to the
+`price` namespace and can be invalidated with one `bump_version("price")` call.
+
+The versions themselves are stored within the cache, within the `_version` namespace.
+(As an implementation detail, this allows one to invalidate _all_ versioned keys by
+bumping the version of `_version`. Very meta!)
+"""
```

Comment:
* Missing license header.
* No need for encoding header.
* Docstrings should be wrapped at col 79 (or I would say preferably around 72 or so, but that's not in our style guide so cannot demand that). -> While wrapping, make the title shorter so it will fit to one line.